### PR TITLE
Fix #7948 Job failure progress messages not coded

### LINKED ISSUE #7948: Job failure progress messages not coded
### How to Reproduce
- Throw an exception with ```getLocalizedMessage``` implementation (e.g. CodedRuntimeException) in a job

### Expected behavior
- Job fails with localized progress message with the language set to the current language of the job owner
- Job log contains the exception message based on ```Exception.getMessage()```

### Observed behavior
- Job fails with progress message based on ```Exception.getMessage()```

### PR DESCRIPTION
**Pull request by dennishendriksen**
#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] (If you have changed REST API interface) view-swagger.ftl updated
- [ ] Test plan template updated
- [ ] Clean commits
- [ ] Added Feature/Fix to release notes

### CODE REVIEW

File: molgenis-jobs/src/main/java/org/molgenis/jobs/JobExecutionTemplate.java [hunk 1]
```diff
@@ -31,36 +33,45 @@
       Progress progress,
       Authentication authentication,
       TransactionOperations transactionOperations) {
-    return runWithAuthentication(
-        authentication, () -> tryCallInTransaction(job, progress, transactionOperations));
+    // FIXME determine locale using job username
```
Comment (tommydeboer) [inline molgenis-jobs/src/main/java/org/molgenis/jobs/JobExecutionTemplate.java]: Resolve `FIXME`

File: molgenis-jobs/src/main/java/org/molgenis/jobs/NontransactionalJob.java [hunk 2]
```diff
@@ -25,6 +26,12 @@ public NontransactionalJob(Progress progress, Authentication authentication) {
 
   @Override
   public T call() {
-    return jobExecutionTemplate.call(this, progress, authentication);
+    // FIXME determine locale using job username
```
Comment (tommydeboer) [inline molgenis-jobs/src/main/java/org/molgenis/jobs/NontransactionalJob.java]: Resolve `FIXME`

File: molgenis-jobs/src/main/java/org/molgenis/jobs/ProgressImpl.java [hunk 3]
```diff
@@ -90,12 +90,12 @@ public void success() {
...
         jobExecution.getLog());
     update();
-    JobExecutionContext.unset();
+    JobExecutionHolder.unset();
   }
 
   @Override
   public void failed(Exception ex) {
-    JOB_EXECUTION_LOG.error("Failed. " + ex.getMessage(), ex);
+    JOB_EXECUTION_LOG.error("Failed. " + ex.getLocalizedMessage(), ex);
```
Comment (tommydeboer) [inline molgenis-jobs/src/main/java/org/molgenis/jobs/ProgressImpl.java]: The error code should be logged as well here.

**Comment by tommydeboer** [comment]
image
Not sure what to expect here, but have you considered showing the localized message above the loading bar?

File: molgenis-jobs/src/main/java/org/molgenis/jobs/ProgressImpl.java (hunk 3 above)
Comment (dennishendriksen) [reply molgenis-jobs/src/main/java/org/molgenis/jobs/ProgressImpl.java]: ```java.lang.Exception``` doesn't have a code

**Comment by dennishendriksen** [comment]
> image
> Not sure what to expect here, but have you considered showing the localized message above the loading bar?

I didn't make changes to this job UI component.

**Comment by tommydeboer** [comment]
> I didn't make changes to this job UI component.

I know, I'm just wondering if you should.

File: molgenis-jobs/src/main/java/org/molgenis/jobs/ProgressImpl.java (hunk 3 above)
Comment (tommydeboer) [reply molgenis-jobs/src/main/java/org/molgenis/jobs/ProgressImpl.java]: The whole point of the custom exceptions was to have the codes visible so we could easily figure out what went wrong, even in cases where it says "エンティティタイプテストの不明な属性テスト". 

I think in cases of `CodedRuntimeException` it should show the code.

Comment (tommydeboer) [reply molgenis-jobs/src/main/java/org/molgenis/jobs/ProgressImpl.java]: Maybe its just weird to log localized exceptions messages in the first place... 🤔

**Pull request merged**